### PR TITLE
Refactor diagnosis agent

### PR DESCRIPTION
**Pull request by samplise**
### What changes were proposed in this pull request?

Refactor diagnosis agent. The agent will observe the problems periodically.

### Why are the changes needed?

Complete the framework that the agent could check if predefined problems happen.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit test

### CODE REVIEW

File: dlrover/python/diagnosis/common/diagnose_action.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+
+class DiagnoseAction:
+    def __init__(self):
+        self.actions: List[str] = []
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/common/diagnose_action.py:19]: ```self._actions``` for protection

File: dlrover/python/diagnosis/inferencechain/coordinator.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from dlrover.python.diagnosis.common.diagnose_action import DiagnoseAction
+from dlrover.python.diagnosis.common.inference_chain import Inference
+
+
+def coordinate_inferences(observations: List[Inference]) -> DiagnoseAction:
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/inferencechain/coordinator.py:20]: What does this mean? Need annotations.

File: dlrover/python/diagnosis/inferencechain/inferenceoperator/collect_metrics_operator.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    InferenceName,
+    InferenceOperator,
+)
+from dlrover.python.diagnosis.datacollector.xpu_timer_metric_collector import (
+    XpuTimerMetricsCollector,
+)
+from dlrover.python.elastic_agent.master_client import MasterClient
+
+
+class CollectMetricsOperator(InferenceOperator):
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/inferencechain/inferenceoperator/collect_metrics_operator.py:32]: ```MetricCollectionOperator```

File: dlrover/python/diagnosis/inferencechain/inferenceoperator/collect_metrics_operator.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    """
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/inferencechain/inferenceoperator/collect_metrics_operator.py:33]: Also i think this operator should not only be used for xpu-timer-collector, but also other collectors.
operator : collector  = 1 : n

File: dlrover/python/diagnosis/inferencechain/coordinator.py (hunk 2 above)
Comment (samplise) [reply dlrover/python/diagnosis/inferencechain/coordinator.py:20]: dlrover may observe multiple problems together. This function is to decide the final actions to resolve those problems.

**Review by BalaBalaYi — APPROVED** [review]
LGTM

**Pull request merged**